src/rgw/rgw_rest_s3.cc

```diff
@@ -3581,7 +3581,7 @@ RGWOp *RGWHandler_REST_Bucket_S3::get_obj_op(bool get_data)
   if (get_data) {   
     if (list_type == 1) {
        return new RGWListBucket_ObjStore_S3;     
-    } else if(list_type == 2) {
+    } else { /* list_type: 2 */
       return new RGWListBucket_ObjStore_S3v2;
     } } else {
```

Comment:
Oops I missed the warning. Thanks for pointing that @iotcg . As of now, we are making listobjectsv1 as the default. So, this won't work. A better approach is:

  if (get_data) {   
    if (list_type == 2) {
       return new RGWListBucket_ObjStore_S3v2;     
    } else {
      return new RGWListBucket_ObjStore_S3;
    }
}
     else {
    return new RGWStatBucket_ObjStore_S3;    
  }

Reply:
There is a similar pr: https://github.com/ceph/ceph/pull/28809
Could you please close this? Really appreciate your effort @iotcg

Reply:
Thanks for your remind. Close now.